backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v1/bpartner/bpartnercomposite/BPartnerCompositeCacheIndex.java

```diff
@@ -138,4 +148,32 @@ public BPartnerId extractDataItemId(@NonNull final BPartnerComposite dataItem)
 			return result;
 		}
 	}
+
+	@Override
```

Comment:
```suggestion
	/**
	  * @return true if the given recordRef is a C_User_Assigned_Role or C_User_Role.
	  *    background: in case of new role being assigned, that new assigned_role_id will not be part of the cached key (as it wasn't there before) thus the cache is not refreshed.
	  */
	@Override
```